Lib/inspect.py

```diff
@@ -326,13 +326,14 @@ def getmembers(object, predicate=None):
     # :dd any DynamicClassAttributes to the list of names if object is a class;
     # this may result in duplicate entries if, for example, a virtual
     # attribute with the same name as a DynamicClassAttribute exists
-    try:
-        for base in object.__bases__:
-            for k, v in base.__dict__.items():
-                if isinstance(v, types.DynamicClassAttribute):
-                    names.append(k)
-    except AttributeError:
-        pass
+    if isclass(object):
```

Comment:
I suggest to move the above comment inside the `if` branch. Seems `:dd` is a typo of `Add`.

This `if isclass(object):` can be merged with other `if isclass(object):` at the beginning of the function.